text_generation/causal_lm/cpp/continuous_batching/library/src/tests/scheduler.cpp

```diff
@@ -0,0 +1,157 @@
+                                                                            GenerationConfig::greedy(), scheduler_config.block_size);
+    SequenceGroup::Ptr sequence_group2 = std::make_shared<SequenceGroup>(1, ov::Tensor(ov::element::i64, {tokens.size()}, tokens.data()),
+                                                                            GenerationConfig::greedy(), scheduler_config.block_size);
+    std::vector<SequenceGroup::Ptr> requests = {sequence_group1, sequence_group2};
+                                                                       
+    
+    // schedule 2 sequence groups that use 4 kv blocks 
+    Scheduler scheduler = Scheduler(scheduler_config);
+    auto out1 = scheduler.schedule(requests);
+
+    std::vector<size_t> ref_ids = {0, 1};
+    EXPECT_EQ(out1.m_scheduled_sequence_groups_ids, ref_ids);
+    EXPECT_EQ(out1.m_block_tables[0].size(), 2);
+    EXPECT_EQ(out1.m_block_tables[1].size(), 2);
+    EXPECT_EQ(out1.m_block_tables[0][0]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[0][0]->get_index(), 0);
+    EXPECT_EQ(out1.m_block_tables[0][1]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[0][1]->get_index(), 1);
+    EXPECT_EQ(out1.m_block_tables[1][0]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[1][0]->get_index(), 2);
+    EXPECT_EQ(out1.m_block_tables[1][1]->is_free(), false);
+    EXPECT_EQ(out1.m_block_tables[1][1]->get_index(), 3);
+    // tokens.size() * 2 tockens should be scheduled on promt phase, corresponding to first two sequences 
+    EXPECT_EQ(out1.m_total_num_scheduled_tokens, tokens.size() * 2);
+    EXPECT_EQ(out1.is_prompt, true);
+    for (auto seq: requests) {
+        seq->finish_iteration();
+    }
+
+
+    SequenceGroup::Ptr sequence_group3 = std::make_shared<SequenceGroup>(2, ov::Tensor(ov::element::i64, {tokens.size()}, tokens.data()),
+                                                                            GenerationConfig::greedy(), scheduler_config.block_size);
+
+    // schedule 1 more sequence group that use 2 kv blocks 
+    std::vector<SequenceGroup::Ptr> requests1 = {sequence_group1, sequence_group2, sequence_group3};
+    auto out2 = scheduler.schedule(requests1);
+
+    std::vector<size_t> ref_ids1 = {2};
+    EXPECT_EQ(out2.m_scheduled_sequence_groups_ids, ref_ids1);
+    EXPECT_EQ(out2.m_block_tables[2].size(), 2);
+    EXPECT_EQ(out2.m_block_tables[2][0]->is_free(), false);
+    EXPECT_EQ(out2.m_block_tables[2][0]->get_index(), 4);
+    EXPECT_EQ(out2.m_block_tables[2][1]->is_free(), false);
+    EXPECT_EQ(out2.m_block_tables[2][1]->get_index(), 5);
+    // tokens.size() tockens should be scheduled on promt phase, corresponding to third sequence
+    EXPECT_EQ(out2.m_total_num_scheduled_tokens, tokens.size()); 
+    for (auto seq: requests1) {
+        seq->finish_iteration();
+    }
+    // at this point we scheduled all available kv blocks
+
+    // sequence_group3 should be evicted
+    auto out3 = scheduler.schedule(requests1);
```

Comment:
can we check that BlockManager does not contain blocks for sequence 3?

Reply:
Added this check.